### ISSUE TITLE
Taudem failed to build; ld: unknown option: -headerpad_max_install_names;

### ISSUE THREAD

**Issue by Kyrish** [OSGeo/homebrew-osgeo4mac#216]
Hello, I've faced with following problem. My OS is 10.12.3

```
$ brew install taudem
==> Installing taudem from osgeo/osgeo4mac
==> Downloading https://github.com/dtarb/TauDEM/archive/a0335e826d579926013e2d1c33c53d413b7c04b3.tar.gz
Already downloaded: /Users/alex/Library/Caches/Homebrew/taudem-5.3.6-dev.tar.gz
==> cmake . -DCMAKE_C_FLAGS_RELEASE=-DNDEBUG -DCMAKE_CXX_FLAGS_RELEASE=-DNDEBUG -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/taudem/5.3.6-dev_1 -DCMAKE_BUILD_TYPE=Release -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_
==> make
Last 15 lines from /Users/alex/Library/Logs/Homebrew/taudem/02.make:
make[1]: *** [CMakeFiles/pitremove.dir/all] Error 2
make[2]: *** [dinfupdependence] Error 1
make[1]: *** [CMakeFiles/dinfupdependence.dir/all] Error 2
ld: unknown option: -headerpad_max_install_names;
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [slopeavedown] Error 1
make[1]: *** [CMakeFiles/slopeavedown.dir/all] Error 2
[ 30%] Linking CXX executable dinfavalanche
/usr/local/Cellar/cmake/3.7.2/bin/cmake -E cmake_link_script CMakeFiles/dinfavalanche.dir/link.txt --verbose=1
/usr/local/Homebrew/Library/Homebrew/shims/super/clang++   -DNDEBUG -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.12.sdk -mmacosx-version-min=10.12 -Wl,-search_paths_first -Wl,-headerpad_max_install_names; -Wl,-flat_namespace  -Wl,-commons,use_dylibs  CMakeFiles/dinfavalanche.dir/DinfAvalanche.cpp.o CMakeFiles/dinfavalanche.dir/DinfAvalanchemn.cpp.o CMakeFiles/dinfavalanche.dir/commonLib.cpp.o CMakeFiles/dinfavalanche.dir/tiffIO.cpp.o  -o dinfavalanche  /usr/local/lib/libmpicxx.dylib /usr/local/lib/libmpi.dylib /usr/local/lib/libpmpi.dylib /usr/local/opt/gdal2/lib/libgdal.dylib 
ld: unknown option: -headerpad_max_install_names;
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [dinfavalanche] Error 1
make[1]: *** [CMakeFiles/dinfavalanche.dir/all] Error 2
make: *** [all] Error 2
```

The problem is semicolon after flag -headerpad_max_install_names , I have no idea how to get rid of it. Eventually I've manually compiled and installed Taudem in DIY-mode by deletion this semicolon in all link.txt followed by $cmake -DCMAKE_INSTALL_PREFIX=/usr/local/Cellar/taudem/5.3.7 ..

(just in case corresponding detailed gist https://gist.github.com/anonymous/0bbd2026dbec5fe86dce67c24c25c735)

**Comment by vanhoesenj** [comment]
Having the same issue

**Comment by Kyrish** [comment]
Thank you for your reply. But I prefer to use mpich for many other things. Anyway I will try.

**Comment by crazyzlj** [comment]
I have encountered the same issue and here is my solution.
Here is part of my CMakeLists.txt:
```cmake
find_package(MPI REQUIRED)
set(CMAKE_CXX_FLAG "${CMAKE_CXX_FLAG} ${MPI_COMPILE_FLAGS}")
set(CMAKE_CXX_LINK_FLAGS "${CMAKE_CXX_LINK_FLAGS} ${MPI_LINK_FLAGS}")
```
What should be aware of is the position of double quotation marks. If delete the "", then a semicolon will be added between two flags. So cause the unrecognized link option.

Hope to help you. Here is [my project as an example](https://github.com/lreis2415/AutoFuzSlpPos/blob/master/autofuzslppos/CMakeLists.txt). CI on Windows (with MSVC 2013 and MSMPI-v8), Linux (with GCC and MPICH2), and macOS (with clang and MPICH2) all succeed. Besides, OpenMPI has been tested on my own computer.

**Comment by nickrobison** [comment]
This has been resolved in #349. Thanks to @crazyzlj for pointing us in the right direction!
